spi_nand_flash/src/spi_nand_oper.c

```diff
@@ -26,7 +26,17 @@ esp_err_t spi_nand_execute_transaction(spi_device_handle_t device, spi_nand_tran
         .dummy_bits = transaction->dummy_bits
     };
 
-    return spi_device_transmit(device, (spi_transaction_t *) &e);
+    if (transaction->flags == SPI_TRANS_USE_TXDATA) {
+        *(uint32_t *)e.base.tx_data = *transaction->mosi_data;
```

Comment:
Recommend an explicit `memcpy`, since we don't know whether `transaction->mosi_data` is 4-byte-aligned. Dereferencing an unaligned pointer with `*(uint32_t*)` may result in an unaligned access exception on some chips.

Reply:
That makes sense. I've updated it.